ibm/service/iampolicy/resource_ibm_iam_policy_template.go

```diff
@@ -0,0 +1,498 @@
+	validateSchema := make([]validate.ValidateSchema, 0)
+	validateSchema = append(validateSchema,
+		validate.ValidateSchema{
+			Identifier:                 "name",
+			ValidateFunctionIdentifier: validate.StringLenBetween,
+			Type:                       validate.TypeString,
+			Required:                   true,
+			MinValueLength:             1,
+			MaxValueLength:             300,
+		},
+		validate.ValidateSchema{
+			Identifier:                 "account_id",
+			ValidateFunctionIdentifier: validate.StringLenBetween,
+			Type:                       validate.TypeString,
+			Required:                   true,
+			MinValueLength:             1,
+			MaxValueLength:             300,
+		},
+		validate.ValidateSchema{
+			Identifier:                 "description",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Optional:                   true,
+			Regexp:                     `^.*$`,
+			MinValueLength:             0,
+			MaxValueLength:             300,
+		},
+	)
+
+	resourceValidator := validate.ResourceValidator{ResourceName: "ibm_iam_policy_template", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIBMIAMPolicyTemplateCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamPolicyManagementClient, err := meta.(conns.ClientSession).IAMPolicyManagementV1API()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createPolicyTemplateOptions := &iampolicymanagementv1.CreatePolicyTemplateOptions{}
+
+	createPolicyTemplateOptions.SetName(d.Get("name").(string))
+	createPolicyTemplateOptions.SetAccountID(d.Get("account_id").(string))
+
+	policyModel, err := generateTemplatePolicy(d.Get("policy.0").(map[string]interface{}), iamPolicyManagementClient)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	createPolicyTemplateOptions.SetPolicy(policyModel)
+	if _, ok := d.GetOk("description"); ok {
+		createPolicyTemplateOptions.SetDescription(d.Get("description").(string))
+	}
+	if _, ok := d.GetOk("committed"); ok {
+		createPolicyTemplateOptions.SetCommitted(d.Get("committed").(bool))
+	}
+
+	policyTemplate, response, err := iamPolicyManagementClient.CreatePolicyTemplateWithContext(context, createPolicyTemplateOptions)
+	if err != nil {
+		log.Printf("[DEBUG] CreatePolicyTemplateWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("CreatePolicyTemplateWithContext failed %s\n%s", err, response))
+	}
+
+	version, _ := strconv.Atoi(*policyTemplate.Version)
+	d.SetId(fmt.Sprintf("%s/%d", *policyTemplate.ID, version))
+	return resourceIBMIAMPolicyTemplateVersionRead(context, d, meta)
+}
+
+func resourceIBMIAMPolicyTemplateUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	_, err := meta.(conns.ClientSession).IAMPolicyManagementV1API()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if d.HasChange("name") || d.HasChange("account_id") {
```

Comment:
Instead of this can we add Force_new : true for name which tells user it wll destroy and recreate

Reply:
we need base template so we can't impose Force_new